src/Microsoft.AspNetCore.Mvc.Core/Routing/MvcEndpointDataSource.cs

```diff
@@ -234,339 +204,10 @@ private DefaultEndpointConventionBuilder ResolveActionConventionBuilder(ActionDe
-                            resolvedRouteValues,
-                            name,
-                            GetPattern(ref patternStringBuilder, newPathSegments),
-                            newPathSegments,
-                            nonInlineDefaultsCopy,
-                            routeOrder++,
-                            dataTokens,
-                            suppressLinkGeneration,
-                            true,
-                            conventions);
-                        endpoints.Add(ep);
-
-                        hasLinkGenerationEndpoint = true;
-                    }
-
-                    var subPathSegments = newPathSegments.Take(i);
-
-                    var subEndpoint = CreateEndpoint(
-                        action,
-                        resolvedRouteValues,
-                        name,
-                        GetPattern(ref patternStringBuilder, subPathSegments),
-                        subPathSegments,
-                        nonInlineDefaultsCopy,
-                        routeOrder++,
-                        dataTokens,
-                        suppressLinkGeneration,
-                        suppressPathMatching,
-                        conventions);
-                    endpoints.Add(subEndpoint);
-                }
-
-                UpdatePathSegments(i, action, resolvedRouteValues, routePattern, newPathSegments, ref allParameterPolicies);
-            }
-
-            var finalEndpoint = CreateEndpoint(
-                action,
-                resolvedRouteValues,
-                name,
-                GetPattern(ref patternStringBuilder, newPathSegments),
-                newPathSegments,
-                nonInlineDefaultsCopy,
-                routeOrder++,
-                dataTokens,
-                suppressLinkGeneration,
-                suppressPathMatching,
-                conventions);
-            endpoints.Add(finalEndpoint);
-
-            return routeOrder;
-
-            string GetPattern(ref StringBuilder sb, IEnumerable<RoutePatternPathSegment> segments)
-            {
-                if (sb == null)
-                {
-                    sb = new StringBuilder();
-                }
-
-                RoutePatternWriter.WriteString(sb, segments);
-                var rawPattern = sb.ToString();
-                sb.Length = 0;
-
-                return rawPattern;
-            }
-        }
-
-        private static IDictionary<string, string> ResolveActionRouteValues(ActionDescriptor action, IReadOnlyDictionary<string, object> allDefaults)
-        {
-            Dictionary<string, string> resolvedRequiredValues = null;
-
-            foreach (var kvp in action.RouteValues)
-            {
-                // Check whether there is a matching default value with a different case
-                // e.g. {controller=HOME}/{action} with HomeController.Index will have route values:
-                // - controller = HOME
-                // - action = Index
```

Comment:
I don't think route pattern transformer is handling this situation. It's a weird edge-case but it is the behavior in ye old routing.